layers/water-material.js

```diff
@@ -0,0 +1,346 @@
+              vec4 foamTex = texture2D(foamTexture, foamUv);
+              foamTex = step(vec4(0.9), foamTex);
+              float foamTextScale = 2.;
+              vec4 foamT = vec4(foamTex.r * (1.0 - foamDepth) * foamTextScale);
+              foamT = mix(vec4(0.), foamT, foamDepth * fadeoutLerp);
+              vec4 foamLineCutOut = saturate(foamT);
+              waterColor = waterColor * ((vec4(1.0) - foamLineCutOut)) + foamT;
+  
+              //################################## handle mirror ##################################
+              vec3 surfaceNormal = normalize(vNormal * vec3(1.5, 1.0, 1.5));
+              vec3 worldToEye = eye - vWorldPosition.xyz;
+              float distance = length(worldToEye);
+              float distortionScale = 3.;
+              vec2 distortion = surfaceNormal.xz * (0.001 + 1.0 / distance) * distortionScale;
+              vec3 reflectionSample = vec3(texture2D(mirror, vUv.xy / vUv.w + distortion));
+              float theta = max(dot(eyePosition, surfaceNormal), 0.0);
+              float rf0 = 0.1;
+              float reflectance = rf0 + (1.0 - rf0) * pow((1.0 - theta), 5.0);
+              float reflectanceScale = 5.;
+              vec3 col1 = reflectionSample * 0.6;
+              vec3 col2 = reflectionSample * 0.2;
+              vec3 albedo = mix(col1, col2, reflectance * reflectanceScale);
+              gl_FragColor = vec4(albedo, waterColor.a);
+              gl_FragColor.rgb += waterColor.rgb;
+            }
+            else{
+              //################################## refraction ##################################
+              vec3 waterColor = vec3(0.126, 0.47628, 0.6048);
+              
+              vec3 surfaceNormal = normalize(vNormal * vec3(1.5, 1.0, 1.5));
+              vec3 worldToEye = eye - vWorldPosition.xyz;
+              float distance = length(worldToEye);
+              float distortionScale = 0.1;
+              vec2 distortion = surfaceNormal.xz * (0.001 + 1.0 / distance) * distortionScale;
+              vec3 reflectionSample = vec3(texture2D(refractionTexture, vUv.xy / vUv.w + distortion));
+              float theta = max(dot(eyePosition, surfaceNormal), 0.0);
+              float rf0 = 0.1;
+              float reflectance = rf0 + (1.0 - rf0) * pow((1.0 - theta), 5.0);
+              float reflectanceScale = 5.;
+              vec3 col1 = reflectionSample * 0.6;
+              vec3 col2 = reflectionSample * 0.2;
+              vec3 albedo = mix(col1, col2, reflectance * reflectanceScale);
+              gl_FragColor = vec4(albedo, 1.0);
+              gl_FragColor.rgb += waterColor.rgb;
+            }
+          }
+          else { // river, for temporary
+            float depthScale = 15.;
+            float depthFalloff = 3.;
+            float sceneDepth = getDepthFade(fragmentLinearEyeDepth, linearEyeDepth, depthScale, depthFalloff);
+
+            vec4 cos_grad = cosine_gradient(sceneDepth, phases, amplitudes, frequencies, offsets);
```

Comment:
lets stay consistent with the style of writing code
so instead of :
`cos_grad` lets do `cosGradient` 
same with the `cosine_gradient` function